vars/commonPipelineEnvironment.groovy

```diff
@@ -192,36 +192,31 @@ class commonPipelineEnvironment implements Serializable {
     ]
 
     void writeToDisk(script) {
-
         files.each({f  ->
-            if (this[f.property] && !script.fileExists(f.filename)) {
-                script.writeFile file: f.filename, text: this[f.property]
-            }
+            writeValueToFile(script, f.filename, this[f.property])
         })
 
         containerProperties.each({key, value ->
-            def fileName = ".pipeline/commonPipelineEnvironment/container/${key}"
-            if (value && !script.fileExists(fileName)) {
-                if(value in CharSequence) {
-                    script.writeFile file: fileName, text: value
-                } else {
-                    script.writeFile file: fileName, text: groovy.json.JsonOutput.toJson(value)
-                }
-            }
+            writeValueToFile(script, ".pipeline/commonPipelineEnvironment/container/${key}", value)
         })
 
         valueMap.each({key, value ->
-            def fileName = ".pipeline/commonPipelineEnvironment/custom/${key}"
-            if (value && !script.fileExists(fileName)) {
-                if(value in CharSequence) {
-                    script.writeFile file: fileName, text: value
-                } else {
-                    script.writeFile file: fileName, text: groovy.json.JsonOutput.toJson(value)
-                }
-            }
+            writeValueToFile(script, ".pipeline/commonPipelineEnvironment/custom/${key}", value)
         })
     }
 
+    void writeValueToFile(script, String filename, value){
+        if (value == null)
+            return
+        if (!value in CharSequence) {
+            filename += '.json'
+            value = groovy.json.JsonOutput.toJson(value)
+        }
+        if (script.fileExists(fileName))
```

Comment:
Why do we write the file only if the file is not already present? How should the behaviour be in case there is already a file, but the data has been changed in memory. Do we want to keep the old (... outdated) content on disk in this case so that the outdated content is used in the go layer? Or do we need to update the content?

In case the current strategy is appropriate: What about moving file file exists check some lines above so that we on't serialize the value for nothing in case the file does already exist.

I understand we had the same strategy earlier, but it really would love to understand why we don't update the file ...